Source/WebGPU/WebGPU/ShaderModule.mm

```diff
@@ -117,7 +166,8 @@
     if (!shaderModuleParameters)
         return ShaderModule::createInvalid(*this);
 
-    auto checkResult = WGSL::staticCheck(fromAPI(shaderModuleParameters->wgsl.code), std::nullopt, { maxBuffersPlusVertexBuffersForVertexStage(), maxBuffersForFragmentStage(), maxBuffersForComputeStage() });
+    auto supportedFeatures = buildFeatureSet(m_capabilities.features);
```

Comment:
We could build this once per device, but it probably doesn't matter as the max feature count is small and could be more expensive to reference the cached version